Source/Core/VideoBackends/Metal/MTLRenderer.mm

```diff
@@ -0,0 +1,502 @@
+      [desc setComputeFunction:fn];
+      [desc setLabel:[fn label]];
+      MRCOwned<id<MTLComputePipelineState>> pipeline =
+          MRCTransfer([g_device newComputePipelineStateWithDescriptor:desc
+                                                              options:MTLPipelineOptionArgumentInfo
+                                                           reflection:&reflection
+                                                                error:&err]);
+      if (err)
+      {
+        DumpBadShader(fmt::format("Failed to compile compute pipeline {}", name));
+        return nullptr;
+      }
+      return std::make_unique<ComputePipeline>(stage, reflection, std::move(msl), std::move(fn),
+                                               std::move(pipeline));
+    }
+    return std::make_unique<Shader>(stage, std::move(msl), std::move(fn));
+  }
+}
+
+std::unique_ptr<NativeVertexFormat>
+Metal::Renderer::CreateNativeVertexFormat(const PortableVertexDeclaration& vtx_decl)
+{
+  @autoreleasepool
+  {
+    return std::make_unique<VertexFormat>(vtx_decl);
+  }
+}
+
+std::unique_ptr<AbstractPipeline>
+Metal::Renderer::CreatePipeline(const AbstractPipelineConfig& config, const void* cache_data,
+                                size_t cache_data_length)
+{
+  return g_object_cache->CreatePipeline(config);
+}
+
+void Metal::Renderer::Flush()
+{
+  @autoreleasepool
+  {
+    g_state_tracker->FlushEncoders();
+  }
+}
+
+void Metal::Renderer::WaitForGPUIdle()
+{
+  @autoreleasepool
+  {
+    g_state_tracker->FlushEncoders();
+    g_state_tracker->WaitForFlushedEncoders();
+  }
+}
+
+void Metal::Renderer::OnConfigChanged(u32 bits)
+{
+  if (bits & CONFIG_CHANGE_BIT_VSYNC)
+    [m_layer setDisplaySyncEnabled:g_ActiveConfig.bVSyncActive];
+
+  if (bits & CONFIG_CHANGE_BIT_ANISOTROPY)
+  {
+    g_object_cache->ReloadSamplers();
+    g_state_tracker->ReloadSamplers();
+  }
+}
+
+void Metal::Renderer::ClearScreen(const MathUtil::Rectangle<int>& rc, bool color_enable,
+                                  bool alpha_enable, bool z_enable, u32 color, u32 z)
+{
+  MathUtil::Rectangle<int> target_rc = Renderer::ConvertEFBRectangle(rc);
+  target_rc.ClampUL(0, 0, m_target_width, m_target_height);
+
+  // All Metal render passes are fullscreen, so we can only run a fast clear if the target is too
+  if (target_rc == MathUtil::Rectangle<int>(0, 0, m_target_width, m_target_height))
+  {
+    // Determine whether the EFB has an alpha channel. If it doesn't, we can clear the alpha
+    // channel to 0xFF. This hopefully allows us to use the fast path in most cases.
+    if (bpmem.zcontrol.pixel_format == PixelFormat::RGB565_Z16 ||
+        bpmem.zcontrol.pixel_format == PixelFormat::RGB8_Z24 ||
+        bpmem.zcontrol.pixel_format == PixelFormat::Z24)
+    {
+      // Force alpha writes, and clear the alpha channel. This is different to the other backends,
+      // where the existing values of the alpha channel are preserved.
```

Comment:
Minor: should be "different from", not "different to"

Reply:
Should I change this in the Vulkan backend where I copied it from as well?

Reply:
Sure, might as well.